torchaudio/prototype/io/streamer.py

```diff
@@ -0,0 +1,486 @@
+        """
+        torch.ops.torchaudio.ffmpeg_streamer_seek(self._s, timestamp)
+
+    def add_basic_audio_stream(
+        self,
+        frames_per_chunk: int,
+        buffer_chunk_size: int = 3,
+        stream_index: Optional[int] = None,
+        sample_rate: Optional[int] = None,
+        dtype: Optional[torch.dtype] = torch.float32,
+    ):
+        """Add output audio stream
+
+        Args:
+            frames_per_chunk (int): A unit number of frames returned by Streamer as a chunk.
+                If the source stream is exhausted before enough frames are buffered,
+                then the chunk is returned as-is.
+
+            buffer_chunk_size (int, optional): Internal buffer size.
+                When this many chunks are created, but
+                client code does not pop the chunk, if a new frame comes in, the old
+                chunk is dropped.
+
+            stream_index (int or None, optional): The source audio stream index.
+                If omitted, :py:attr:`default_audio_stream` is used.
+
+            sample_rate (int or None, optional): If provided, resample the audio.
+
+            dtype (torch.dtype, optional): If not ``None``, change the output sample precision.
+                If floating point, then the sample value range is
+                `[-1, 1]`.
+        """
+        i = self.default_audio_stream if stream_index is None else stream_index
+        torch.ops.torchaudio.ffmpeg_streamer_add_basic_audio_stream(
+            self._s, i, frames_per_chunk, buffer_chunk_size, sample_rate, dtype
+        )
+
+    def add_basic_video_stream(
+        self,
+        frames_per_chunk: int,
+        buffer_chunk_size: int = 3,
+        stream_index: Optional[int] = None,
+        frame_rate: Optional[int] = None,
+        width: Optional[int] = None,
+        height: Optional[int] = None,
+        format: Optional[str] = "RGB",
+    ):
+        """Add output video stream
+
+        Args:
+            frames_per_chunk (int): A unit number of frames returned by Streamer as a chunk.
+                If the source stream is exhausted before enough frames are buffered,
+                then the chunk is returned as-is.
+
+            buffer_chunk_size (int, optional): Internal buffer size.
+                When this many chunks are created, but
+                client code does not pop the chunk, if a new frame comes in, the old
+                chunk is dropped.
+
+            stream_index (int or None, optional): The source video stream index.
+                If omitted, :py:attr:`default_video_stream` is used.
+
+            frame_rate (int or None, optional): If provided, change the frame rate.
+
+            width (int or None, optional): If provided, change the image width. Unit: Pixel.
+            height (int or None, optional): If provided, change the image height. Unit: Pixel.
+            format (str, optional): Change the format of image channels. Valid values are,
```

Comment:
`None` for these parameters are just placeholder for "missing value that is to be fallback to some (dynamically determined) default value". So `None` is not what we expect / advertise users to pass. 

Consistency is important but I think saying `or None` is more confusing because the code expect that users to pass some meaningful value other than `None` if users are to provide. What do you think? @hwangjeff  @carolineechen

Reply:
ah okay, I misunderstood what you meant.

Reply:
ah, one more nit: it seems like we have a convention of adding default values for kwargs in docstrings (this applies throughout the module)